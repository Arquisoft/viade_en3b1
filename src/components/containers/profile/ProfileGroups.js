import React, { useState, useEffect } from 'react';
import { Container, Typography, makeStyles, CssBaseline, Grid, Paper, Link, Backdrop, CircularProgress } from '@material-ui/core';
import NavBar from '../../ui/main/NavBar';
import Footer from '../../ui/main/Footer';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import GroupsCache from '../../../cache/GroupsCache';
import GroupCard from '../../ui/GroupCard';

export default function ProfileGroups() {

    const classes = useStyles();

    const [groups, setGroups] = useState([]);
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        GroupsCache.getGroups().then((gList) => {
            let groupsList = [];
            // gList.forEach((g) => {
            //     groupsList.push(g);
            //     console.log(g)
            // })
            groupsList.push(groupsList);
            setGroups(gList);
            setLoading(false);
        })
    }, [])

    return (
        <div className={classes.root}>
            <NavBar />
            <CssBaseline />
            <Container
                component="main"
                className={classes.main}
                maxWidth="md"
            >
                <Typography variant="h3">Your Groups</Typography>

                <Paper
                    elevation={0}
                    className={classes.paper}>
                    <Tabs
                        value={2}
                        indicatorColor="primary"
                        textColor="primary"
                    >
                        <Tab selected={false} label="General" href="#/profile" />
                        <Tab selected={true} label="Friends" href="#/profile/friends" />
                        <Tab label="Groups" href="#/profile/groups" />
                    </Tabs>
                </Paper>

                {(loading) ? (
                    <div style={{ height: '30rem' }}>
                        <Backdrop style={{ color: '#5c5585' }} open={loading} invisible>
                            <CircularProgress color="inherit" hidden={loading} />
                        </Backdrop>
                    </div>
                ) : (
                        (groups) ? (
                            <GroupsList groups={groups} />
                        ) : (
                                <DefaultPlaceHolder />
                            )
                    )}

            </Container>

            <Footer />
        </div>
    )
}

function DefaultPlaceHolder() {
    return (
        <div style={{ textAlign: 'center', marginTop: '15rem', marginBottom: '15rem' }}>
            <Typography variant="h6">You don't have any groups yet.</Typography>
            <Typography variant="h6">Want to {<Link style={{ color: "#7c4dff" }} href={"#/profile/groups/create"}>add</Link>} some?</Typography>
        </div>
    );
}

function GroupsList(props) {
    const { groups } = props;
    const classes = useStyles();

    return (
        <Grid
            item
            xs={false}
            sm={12}
            md={12}
            className={classes.groups}
        >

            <main className={classes.layout}>
                    {groups.map((each, index) => (
                        <div key={index}>
                            <GroupCard group={each} />
                        </div>
                    ))}
            </main>

        </Grid>
    );
}

const useStyles = makeStyles((theme) => ({
    root: {
        display: 'flex',
        flexDirection: 'column',
    },
    main: {
        marginTop: theme.spacing(8),
        marginBottom: theme.spacing(2),
    },
    paper: {
        borderBottom: '1px solid #e8e8e8',
        marginTop: theme.spacing(2),
        backgroundColor: "#fafafa",
        width: '100%'
    },
    layout: {
        width: 'auto',
        marginLeft: theme.spacing(2),
        marginRight: theme.spacing(2),
        [theme.breakpoints.up(600 + theme.spacing(2) * 2)]: {
            width: 600,
            marginLeft: 'auto',
            marginRight: 'auto',
        },
    },
    paper2: {
        marginTop: theme.spacing(3),
        marginBottom: theme.spacing(3),
        padding: theme.spacing(2),
        backgroundColor: "#fafafa",
        [theme.breakpoints.up(600 + theme.spacing(3) * 2)]: {
            marginTop: theme.spacing(6),
            marginBottom: theme.spacing(6),
            padding: theme.spacing(3),
        },
    },
    friends: {
        marginTop: theme.spacing(5),
    },
    fPhoto: {
        width: theme.spacing(13),
        height: theme.spacing(13),
        marginLeft: '3rem',
        marginTop: '1rem'
    },
    fCard: {
        display: 'block',
        width: '15rem',
        height: theme.spacing(30)
    },
    friendsList: {
        textAlign: 'center',
        marginTop: theme.spacing(7),
        marginBottom: theme.spacing(50)
    }
}));

// export default withStyles(useStyles)(ProfileGroups);