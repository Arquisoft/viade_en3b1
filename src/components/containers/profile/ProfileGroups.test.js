import ProfileGroups from './ProfileGroups';

test('Define ProfileGroups', () => {
  expect(ProfileGroups).toBeDefined();
});

test('Renders correctly', () => {
  const wrapper = render(<ProfileGroups />);
  expect(wrapper).toMatchSnapshot();
});