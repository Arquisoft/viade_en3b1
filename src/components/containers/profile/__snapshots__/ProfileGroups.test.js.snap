// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  class="makeStyles-root-1"
>
  <div
    class="makeStyles-root-10"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-11 MuiToolbar-gutters"
      >
        <div
          class="MuiGrid-root makeStyles-link-13"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
              href="#/home"
            >
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <img
                    alt="Viade Logo"
                    src="logo_2.svg"
                    style="height:2.5rem;width:2.5rem"
                  />
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4"
                  >
                    ViaDe
                  </h4>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>
  <main
    class="MuiContainer-root makeStyles-main-2 MuiContainer-maxWidthMd"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3"
    >
      Your Groups
    </h3>
    <div
      class="MuiPaper-root makeStyles-paper-3 MuiPaper-elevation0 MuiPaper-rounded"
    >
      <div
        class="MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed"
          style="overflow:hidden"
        >
          <div
            class="MuiTabs-flexContainer"
            role="tablist"
          >
            <a
              aria-disabled="false"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
              href="#/profile"
              role="tab"
              tabindex="0"
            >
              <span
                class="MuiTab-wrapper"
              >
                General
              </span>
            </a>
            <a
              aria-disabled="false"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
              href="#/profile/friends"
              role="tab"
              tabindex="0"
            >
              <span
                class="MuiTab-wrapper"
              >
                Friends
              </span>
            </a>
            <a
              aria-disabled="false"
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
              href="#/profile/groups"
              role="tab"
              tabindex="0"
            >
              <span
                class="MuiTab-wrapper"
              >
                Groups
              </span>
              <span
                class="PrivateTabIndicator-root-228 PrivateTabIndicator-colorPrimary-229 MuiTabs-indicator"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      style="height:30rem"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiBackdrop-invisible"
        style="opacity:0;color:#5c5585"
      >
        <div
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate"
          hidden=""
          role="progressbar"
          style="width:40px;height:40px"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
    </div>
  </main>
  <div>
    <footer
      class="makeStyles-footer-248"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          ViaDe En3B1
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
        >
          Made by Pablo Ca√±al
        </h6>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="https://github.com/Arquisoft/viade_en3b1/"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
            />
          </svg>
        </a>
      </div>
    </footer>
  </div>
</div>
`;
